File: 백준/Gold/11265. 끝나지 않는 파티/끝나지 않는 파티.py
N, M = map(int, input().split())
party = []
for i in range(N):
    party.append(list(map(int,input().split())))
    
# 3중 for문을 돌면서
for a in range(N):
    for b in range(N):
        for c in range(N):
            # 만약 b->c 직행보다 a를 거쳐가는게 더 빠르면 교체
            if party[b][c] > party[b][a] + party[a][c]:
                party[b][c] = party[b][a] + party[a][c]

for _ in range(M):
    a, b, time = map(int, input().split())
    a -= 1
    b -= 1
    # 남은 시간과 최적화해둔 루트를 확인
    if party[a][b] <= time:
        print('Enjoy other party')
    else:
        print('Stay here')

File: 백준/Gold/11265. 끝나지 않는 파티/끝나지 않는 파티.java

import java.util.*;

public class Main {
    public static void main(String[] args) throws Exception{

        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int m = sc.nextInt();
        int[][] party = new int[n][n];
        for(int i = 0; i < n; i++){
            for(int j = 0; j < n; j++){
                party[i][j] = sc.nextInt();
            }
        }

        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                for (int k = 0; k < n; k++) {
                    if (party[j][k] > party[j][i] + party[i][k]) {
                        party[j][k] = party[j][i] + party[i][k];
                    }
                }
            }
        }

        for (int i = 0; i < m; i++) {
            int a = sc.nextInt();
            int b = sc.nextInt();
            int time = sc.nextInt();
            a -= 1;
            b -= 1;
            if (party[a][b] <= time) {
                System.out.println("Enjoy other party");

            } else {
                System.out.println("Stay here");
            }

        }
    }
}
